nixos/modules/services/monitoring/grafana.nix

```diff
@@ -377,390 +268,945 @@ in {
+
+                muteTimes = [{
+                  orgId = 1;
+                  name = "mti_1";
+                  time_intervals = [{
+                    times = [{
+                      start_time = "06:00";
+                      end_time = "23:59";
+                    }];
+                    weekdays = [
+                      "monday:wednesday"
+                      "saturday"
+                      "sunday"
+                    ];
+                    months = [
+                      "1:3"
+                      "may:august"
+                      "december"
+                    ];
+                    years = [
+                      "2020:2022"
+                      "2030"
+                    ];
+                    days_of_month = [
+                      "1:5"
+                      "-3:-1"
+                    ];
+                  }];
+                }];
+
+                deleteMuteTimes = [{
+                  orgId = 1;
+                  name = "mti_1";
+                }];
+              }
+            '';
+          };
+        };
       };
     };
-
-    extraOptions = mkOption {
-      description = lib.mdDoc ''
-        Extra configuration options passed as env variables as specified in
-        [documentation](http://docs.grafana.org/installation/configuration/),
-        but without GF_ prefix
-      '';
-      default = {};
-      type = with types; attrsOf (either str path);
-    };
   };
 
   config = mkIf cfg.enable {
     warnings = flatten [
+      #(optional (
+      #  cfg.settings.database.password != opt.settings.database.password.default ||
+      #  cfg.settings.security.admin_password != opt.settings.security.admin_password.default
+      #) "Grafana passwords will be stored as plaintext in the Nix store!")
       (optional (
-        cfg.database.password != opt.database.password.default ||
-        cfg.security.adminPassword != opt.security.adminPassword.default
-      ) "Grafana passwords will be stored as plaintext in the Nix store!")
-      (optional (
-        any (x: x.password != null || x.basicAuthPassword != null || x.secureJsonData != null) cfg.provision.datasources
+        let
+          checkOpts = opt: any (x: x.password != null || x.basicAuthPassword != null || x.secureJsonData != null) opt;
+          datasourcesUsed = if (cfg.provision.datasources.settings == null) then [] else cfg.provision.datasources.settings.datasources;
+        in if (builtins.isList cfg.provision.datasources) then checkOpts cfg.provision.datasources else checkOpts datasourcesUsed
       ) "Datasource passwords will be stored as plaintext in the Nix store!")
       (optional (
         any (x: x.secure_settings != null) cfg.provision.notifiers
       ) "Notifier secure settings will be stored as plaintext in the Nix store!")
+      (optional (
+        builtins.isList cfg.provision.datasources
+      ) ''
+          Provisioning Grafana datasources with options had been deprecated.
```

Comment:
That's right. Another option would be `was deprecated` I guess, but it looks like Nixpkgs has an unofficial convention to use `has been deprecated`.

Reply:
> Another option would be was deprecated I guess, but it looks like Nixpkgs has an unofficial convention to use has been deprecated.

Not sure, the details regarding which past tenses to use in what case where an arch enemy of mine when learning English :D 

OK, I actually did some research now: strictly speaking, `was deprecated` seems more correct: `has been` is "present perfect continuous" and describes an action that starts in the past and is still going on while `was` is simple past continuous and describes an action that started and ended in the past. Since you already pushed the change, the deprecation happened entirely in the past :upside_down_face: 

(Not going to unresolve this thread since the current wording is consistent, at least :p)